crates/biome_js_analyze/src/lint/nursery/no_useless_undefined.rs

```diff
@@ -0,0 +1,447 @@
+    /// function foo() {
+    ///    return undefined;
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// function* foo() {
+    ///   yield undefined;
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// function foo(bar = undefined) {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// function foo({bar = undefined}) {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// foo(undefined);
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// let foo;
+    /// const {foo} = bar;
+    /// function foo() {
+    ///   return;
+    /// }
+    /// function* foo() {
+    ///   yield;
+    /// }
+    /// function foo(bar) {}
+    /// function foo({bar}) {}
+    /// foo();
+    /// ```
+    ///
+    pub NoUselessUndefined {
+        version: "next",
+        name: "noUselessUndefined",
+        language: "js",
+        fix_kind: FixKind::Unsafe,
+        sources: &[RuleSource::EslintUnicorn("no-useless-undefined")],
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub RuleQuery = JsVariableStatement
+        // | JsObjectBindingPattern
+        | JsObjectBindingPatternShorthandProperty
+        | JsYieldArgument
+        | JsReturnStatement
+        | JsArrayBindingPatternElement
+        | JsCallExpression
+        | JsArrowFunctionExpression
+        | JsFormalParameter
+}
+
+static FUNCTION_NAMES: [&str; 28] = [
+    // Function#bind()
+    "bind",
+    // Compare function names
+    "is",
+    "equal",
+    "notEqual",
+    "strictEqual",
+    "notStrictEqual",
+    "propertyVal",
+    "notPropertyVal",
+    "not",
+    "include",
+    "property",
+    "toBe",
+    "toHaveBeenCalledWith",
+    "toContain",
+    "toContainEqual",
+    "toEqual",
+    "same",
+    "notSame",
+    "strictSame",
+    "strictNotSame",
+    // `array.push(undefined)`
+    "push",
+    // `array.unshift(undefined)`
+    "unshift",
+    // `array.includes(undefined)`
+    "includes",
+    // `set.add(undefined)`
+    "add",
+    // `set.has(undefined)`
+    "has",
+    // `map.set(foo, undefined)`
+    "set",
+    // `React.createContext(undefined)`
+    "createContext",
+    // https://vuejs.org/api/reactivity-core.html#ref
+    "ref",
+];
+
+fn should_ignore_function(expr: &AnyJsExpression) -> bool {
+    let name = match expr {
+        AnyJsExpression::JsIdentifierExpression(ident) => ident.text(),
+        AnyJsExpression::JsStaticMemberExpression(member_expr) => {
+            let member = member_expr.member().ok().unwrap();
+            member.text()
+        }
+        _ => return false,
+    };
+
+    FUNCTION_NAMES.contains(&name.as_str()) ||
+    // setState(undefined), setXXX(undefined)
+    name.starts_with("set")
+}
+
+fn is_undefined(expr: Option<&AnyJsExpression>) -> Option<(String, TextRange)> {
+    let ident = expr?.as_js_reference_identifier()?;
+    if ident.is_undefined() {
+        Some((ident.text().to_string(), ident.range()))
```

Comment:
Just wondering: why do we need the string that contains `undefined`? Is the range not enough?

If we need both, then we could simply return the toekn (`JsSyntaxToken`).

Reply:
Right, I've followed the `noUselessUndefinedInitialization` implementation, but the text isn't needed. Thank you.

Reply:
Sorry, I was mistaken. Here, it uses identifer to look for the target binding to remove the initializer.

Additional changes:.
Since there doesn't seem to be much need to keep trivia here, I changed it to simply remove the initializer of a statement that uses undefined.